src/teleop.rs

```diff
@@ -0,0 +1,75 @@
+use crate::config::TeleopConfig;
+use rosrust;
+use rosrust_msg;
+use std::collections::HashMap;
+use termion::event::Key;
+
+pub struct Teleoperator {
+    current_velocities: Velocities,
+    _cmd_val_pub: rosrust::Publisher<rosrust_msg::geometry_msgs::Twist>,
+    _keys_to_direction: HashMap<Key, (String, i8)>,
+    _increment: f64,
+}
+
+pub struct Velocities {
+    x: f64,
+    y: f64,
+    theta: f64,
+}
+
+fn match_key_to_string(
+    string_to_direction: &HashMap<String, (String, i8)>,
+) -> HashMap<Key, (String, i8)> {
+    let mut keys_to_direction = HashMap::new();
+    for (string, val) in string_to_direction {
+        keys_to_direction.insert(Key::Char(string.chars().next().unwrap()), val.clone());
+    }
+    keys_to_direction
+}
+
+impl Teleoperator {
+    pub fn new(config: TeleopConfig) -> Teleoperator {
+        let cmd_val_publisher = rosrust::publish(&config.cmd_vel_topic, 1).unwrap();
+        let initial_velocities = Velocities {
+            x: 0.,
+            y: 0.,
+            theta: 0.,
+        };
+        Teleoperator {
+            _cmd_val_pub: cmd_val_publisher,
+            current_velocities: initial_velocities,
+            _keys_to_direction: match_key_to_string(&config.key_mapping),
+            _increment: config.increment.clone(),
+        }
+    }
+
+    pub fn handle_input_from_key(&mut self, input: Key) {
```

Comment:
would it hurt to use the same key mappings for the pose estimate? that would make the whole app more consistent

Reply:
I think that would be fine. This should be done when moving the pose sender to a class

Reply:
@carzum I agree with you. This would also make the config file way less tedious (assuming 5 modes and configuring keys for each one is annoying. Same with help page). However, not that a/d will then become for strifing and q/e for rotation. As a gamer I like it even better, but for non-omnidirectional robot it would feel a bit weird...

Reply:
I tried a little and it doesn't feel weird after some moments,  the middle finger will just prod w and s, the index and ring finger don't care about qe or ad after a few seconds